zuul/connection/github.py

```diff
@@ -566,6 +566,10 @@ def getPullBySha(self, sha):
 
         if len(pulls) > 1:
             raise Exception('Multiple pulls found with head sha %s' % sha)
+            for pull in pulls:
```

Comment:
This code wont actually run because the Exception will already have been raised at this point